### PR TITLE
index planning: use planning for all sizeable blocks

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
We had a patch (#12608) to prevent using stale stats. That wasn't effective and still left the possibility of calculating stats on an empty head, the head receiving series and then using the empty stats on a non-empty head. This results in very inefficient plans. Now that we have #12604, we can remove this logic.

This PR makes the logic for when we use planning a bit more testable with the `lookupplan.PlannerFactory` and removed the changes from #12608

related to #12432 and #12604

### CODE REVIEW

File: pkg/ingester/lookupplan/planner_factory.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+// For very small blocks (< 10,000 series), it returns NoopPlanner to avoid planning overhead.
+// For larger blocks, it generates statistics and returns a CostBasedPlanner.
+// If statistics generation fails, it falls back to NoopPlanner.
+func (p *PlannerFactory) CreatePlanner(meta tsdb.BlockMeta, reader tsdb.IndexReader) index.LookupPlanner {
+	logger := log.With(p.logger, "block", meta.ULID.String(), "block_series", meta.Stats.NumSeries)
+	const statsThreshold = 10_000
+
+	if meta.Stats.NumSeries < statsThreshold {
+		// For very small blocks, the planning overhead is likely not worth it.
+		// This also prevents problems when we've gathereed stats on an empty head block,
```
Comment (chencs) [inline pkg/ingester/lookupplan/planner_factory.go]:
```suggestion
		// This also prevents problems when we've gathered stats on an empty head block,
```

File: pkg/ingester/lookupplan/planner_factory.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		// but then the ingester statrs receiving series for that tenant.
```
Comment (chencs) [inline pkg/ingester/lookupplan/planner_factory.go]:
```suggestion
		// but then the ingester starts receiving series for that tenant.
```

File: pkg/ingester/lookupplan/planner_factory.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	}
+}
+
+// CreatePlanner returns an appropriate index.LookupPlanner for the given block metadata and reader.
+// For very small blocks (< 10,000 series), it returns NoopPlanner to avoid planning overhead.
+// For larger blocks, it generates statistics and returns a CostBasedPlanner.
+// If statistics generation fails, it falls back to NoopPlanner.
+func (p *PlannerFactory) CreatePlanner(meta tsdb.BlockMeta, reader tsdb.IndexReader) index.LookupPlanner {
+	logger := log.With(p.logger, "block", meta.ULID.String(), "block_series", meta.Stats.NumSeries)
+	const statsThreshold = 10_000
```
Comment (chencs) [inline pkg/ingester/lookupplan/planner_factory.go]: It would be nice (if we need to tune this) to pull this out to the top-level, and then have the boundary tests (e.g., "medium 5000-series block," "boundary case just below threshold (9999 series)", at the boundary, etc.) set those series with arithmetic based on the constant instead of hardcoded.
